myning/objects/macguffin.py

```diff
@@ -1,27 +1,71 @@
 from myning.objects.object import Object
+from myning.objects.singleton import Singleton
+from myning.utils.file_manager import FileManager
 
 
-class Macguffin(Object):
-    def __init__(self, exp_boost=1, mineral_boost=1):
-        self.exp_boost = exp_boost
+class Macguffin(Object, metaclass=Singleton):
+    @classmethod
+    def initialize(cls):
+        macguffin = FileManager.load(Macguffin, cls.file_name)
+        if not macguffin:
+            macguffin = cls()
+        cls._instance = macguffin
+
+    @classmethod
+    @property
+    def file_name(cls):
+        return "macguffin"
+
+    def __init__(
+        self,
+        xp_boost=1,
+        mineral_boost=1,
+        research_boost=1,
+        soul_credit_boost=1,
+    ):
+        self.xp_boost = xp_boost
         self.mineral_boost = mineral_boost
+        self.research_boost = research_boost
+        self.soul_credit_boost = soul_credit_boost
 
     @classmethod
     def from_dict(cls, data) -> "Macguffin":
         if data is None:
             return Macguffin()
-        return Macguffin(data["exp_boost"], data["store_boost"])
+        return Macguffin(
+            data["exp_boost"],
```

Comment:
we need mineral boost in here and in to_dict?

Reply:
Unfortunately, I named the mineral boost "store_boost" and I don't want to migrate the keys. I'll add a comment.

Reply:
I feel like this is the ideal time to fix the key since the macguffin object is getting migrated, although it technically doesn't hurt anything